hamilflow/models/brownian_motion.py

```diff
@@ -0,0 +1,205 @@
+from functools import cached_property
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+from pydantic import BaseModel, computed_field, field_validator
+
+
+class BrownianMotionSystem(BaseModel):
+    r"""Definition of the Brownian Motion system
+
+    For consistency, we always use
+    $\mathbf x$ for displacement, and
+    $t$ for steps. The model we are using is
+
+    $$
+    \begin{align}
+    \mathbf x(t + \mathrm dt) &= \mathrm x(t) +
+    \mathcal{N}(\mu=0, \sigma=\sigma \sqrt{\mathrm d t})
+    \end{align}
+    $$
+
+    References:
+
+    1. Brownian motion and random walks. [cited 13 Mar 2024].
+        Available: https://web.mit.edu/8.334/www/grades/projects/projects17/OscarMickelin/brownian.html
+    2. Contributors to Wikimedia projects. Brownian motion.
+        In: Wikipedia [Internet]. 22 Jan 2024 [cited 13 Mar 2024].
+        Available: https://en.wikipedia.org/wiki/Brownian_motion
+
+    :param sigma: base standard deviation
+        to be used to compute the variance
+    :param delta_t: time granunality of the motion
+    """
+
+    sigma: float
+    delta_t: float = 1
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def gaussian_scale(self) -> float:
+        """The scale (standard deviation) of the Gaussian term
+        in Brownian motion
+        """
+        return self.sigma**2 * self.delta_t
+
+    @field_validator("sigma", "delta_t")
+    @classmethod
+    def check_sigma_delta_t_non_negative(cls, v: float) -> float:
```

Comment:
How about `sigma: float = Field(ge=0)`

Reply:
tldr: Nice. This is cleaner. I will switch to Field.

----

I actually designed it twice and one of the versions what using Field.

However, I thought that the error message doesn't say the current value. This is why I wanted to have a customized error message that tells the error and the the current value. 

Example:

Using Field(ge=0)

```
BrownianMotionSystem
sigma
  Input should be greater than or equal to 0 [type=greater_than_equal, input_value=-1, input_type=int]
```

The current validator

```
sigma
  Value error, Value of sigma and delta_t should be positive: v=-1.0 [type=value_error, input_value=-1, input_type=int]
```

Now as I write the reply, I just realized that I completely overlooked the part in `[type=value_error, input_value=-1, input_type=int]`...

How blind was I....